Cannot build with `error[E0308]: mismatched types` in  azul-text-layout using x86_64-pc-windows-msvc toolchain

## Description

Some part of Azul's build system is broken maybe. I tried build Azul with a minimal bin-crate but I got these 9 errors in belows:

Pattern 1: expected `i32`, found `i64` ( x3 errors )

```sh
C:\Users\usagi\.cargo\git\checkouts\azul-f3b1e21a5e685cad\17b76b0\cargo\azul-text-layout\../../azul-text-layout\text_shaping.rs:309:73
    |
309 |         let error = FT_New_Memory_Face(ft_library, font_bytes.as_ptr(), buf_len, font_index as i64, &mut ft_face);
    |                                                                         ^^^^^^^ expected `i32`, found `i64`
    |
help: you can convert an `i64` to `i32` and panic if the converted value wouldn't fit
    |
309 |         let error = FT_New_Memory_Face(ft_library, font_bytes.as_ptr(), buf_len.try_into().unwrap(), font_index as i64, &mut 
ft_face);
    |                                                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Pattern 2: expected `i64`, found `i32` ( x6 erroes )

```sh
error[E0308]: mismatched types
   --> 
C:\Users\usagi\.cargo\git\checkouts\azul-f3b1e21a5e685cad\17b76b0\cargo\azul-text-layout\../../azul-text-layout\text_shaping.rs:333:22
    |
333 |             x_scale: metrics.x_scale,
    |                      ^^^^^^^^^^^^^^^ expected `i64`, found `i32`
```

And, you can see the all errors in here(gist) -> https://gist.github.com/usagi/cf130471216f05aaaf3eed93bd78a462

## Version / OS

* azul version: 17b76b064459e88e8d30904b5548da4b5ed38f76 ( `azul = { git = "https://github.com/maps4print/azul" }` )
* Operating system: Microsoft Windows 10 ( 64-bit ) 10.0.19041.113

## Steps to Reproduce

1. `cargo new a`
2. `cd a`
3. Append line `azul = { git = "https://github.com/maps4print/azul" }` in the `[dependencies]` section in the `Cargo.toml`.
4. `cargo build`

## Additional Information

I think, it is very similar problem in these:

1. https://github.com/maps4print/azul/issues/203#issuecomment-526956322
2. https://github.com/maps4print/azul/issues/163#issuecomment-586701643

But #203 and #163 is the Harfbuzz issue and it's cleared at least the main problems. And, I tried add the msvc's `cmake.exe` path to `PATH` environment, and checked using `where.exe`, and removed the other CMake installs but the problem is not solved. 😢

--- a/azul-text-layout/text_shaping.rs
+++ b/azul-text-layout/text_shaping.rs
@@ -272,12 +272,13 @@ pub fn get_font_metrics_freetype(font_bytes: &[u8], font_index: i32) -> FontMetr
 
     use std::convert::TryInto;
     use freetype::freetype::{
+        FT_Long, FT_F26Dot6,
         FT_Init_FreeType, FT_Done_FreeType, FT_New_Memory_Face,
         FT_Done_Face, FT_Set_Char_Size, FT_Library, FT_Face,
     };
 
     const FT_ERR_OK: i32 = 0;
-    const FAKE_FONT_SIZE: i64 = 1000;
+    const FAKE_FONT_SIZE: FT_F26Dot6 = 1000;
 
     let mut baseline = FontMetrics {
         font_size: FAKE_FONT_SIZE as usize,
@@ -291,7 +292,7 @@ pub fn get_font_metrics_freetype(font_bytes: &[u8], font_index: i32) -> FontMetr
         max_advance: 0,
     };
 
-    let buf_len: i64 = match font_bytes.len().try_into().ok() {
+    let buf_len: FT_Long = match font_bytes.len().try_into().ok() {
         Some(s) => s,
         None => return baseline, // font too large for freetype
     };
@@ -306,7 +307,7 @@ pub fn get_font_metrics_freetype(font_bytes: &[u8], font_index: i32) -> FontMetr
 
         // Load font
         let mut ft_face: FT_Face = ptr::null_mut();
-        let error = FT_New_Memory_Face(ft_library, font_bytes.as_ptr(), buf_len, font_index as i64, &mut ft_face);
+        let error = FT_New_Memory_Face(ft_library, font_bytes.as_ptr(), buf_len, font_index as FT_Long, &mut ft_face);
         if error != FT_ERR_OK {
             FT_Done_FreeType(ft_library);
             return baseline;
@@ -330,12 +331,12 @@ pub fn get_font_metrics_freetype(font_bytes: &[u8], font_index: i32) -> FontMetr
             font_size: FAKE_FONT_SIZE as usize,
             x_ppem: metrics.x_ppem,
             y_ppem: metrics.y_ppem,
-            x_scale: metrics.x_scale,
-            y_scale: metrics.y_scale,
-            ascender: metrics.ascender,
-            descender: metrics.descender,
-            height: metrics.height,
-            max_advance: metrics.max_advance,
+            x_scale: metrics.x_scale as i64,
+            y_scale: metrics.y_scale as i64,
+            ascender: metrics.ascender as i64,
+            descender: metrics.descender as i64,
+            height: metrics.height as i64,
+            max_advance: metrics.max_advance as i64,
         };
 
         FT_Done_Face(ft_face);

